tests/results/test_result_factory.py

```diff
@@ -84,7 +90,77 @@ def foo():
     assert result_factory.persist_result is True
 
 
-def test_roto_flow_can_opt_out_when_persist_result_default_is_overriden_by_setting():
+async def test_root_flow_default_remote_storage():
+    @flow
+    async def foo():
+        result_fac = get_run_context().result_factory
+        return result_fac.storage_block
+
+    block = LocalFileSystem()
+    await block.save("my-result-storage")
+
+    with temporary_settings(
+        {
+            PREFECT_RESULTS_PERSIST_BY_DEFAULT: True,
+            PREFECT_DEFAULT_RESULT_STORAGE_BLOCK: "local-file-system/my-result-storage",
+        }
+    ):
+        storage_block = await foo()
+
+    assert_blocks_equal(storage_block, block)
+    assert storage_block._is_anonymous is False
+
+
+async def test_root_flow_default_remote_storage_saves_correct_result():
+    import base64
+    import pickle
+
+    await LocalFileSystem(basepath="~/.prefect/results").save("my-result-storage")
+
+    @task(result_storage_key="my-result.pkl")
+    async def bar():
+        return {"foo": "bar"}
+
+    @flow
+    async def foo():
+        return await bar()
+
+    with temporary_settings(
+        {
+            PREFECT_RESULTS_PERSIST_BY_DEFAULT: True,
+            PREFECT_DEFAULT_RESULT_STORAGE_BLOCK: "local-file-system/my-result-storage",
+        }
+    ):
+        result = await foo()
+
+    assert result == {"foo": "bar"}
+    local_storage = await LocalFileSystem.load("my-result-storage")
+    result_bytes = await local_storage.read_path("~/.prefect/results/my-result.pkl")
+    saved_python_result = pickle.loads(
+        base64.b64decode(PersistedResultBlob.parse_raw(result_bytes).data)
+    )
+    assert saved_python_result == {"foo": "bar"}
```

Comment:
is there a better way to go from `storage_result_key` -> `saved_python_result`?

Reply:
> if you initialize a `PersistedResult`

it seems that you need to provide the actual python `obj` in order to call `PersistedResult.create`, which I was thinking should be done on my behalf when I run the flow with the new enabled setting

after the running the flow, I'd like to prove that the object my task returned is in fact stored where the result storage setting + my result storage key says it is - so I'm not sure I can use `PersistedResult` here, does that track?